src/SmartComponents/Inventory/applications/compliance/ComplianceRemediationButton.js

```diff
@@ -2,60 +2,91 @@ import React from 'react';
+        });
+    }
+
+    removeRefIdPrefix = (ref_id) => {
+        return ref_id.split('xccdf_org.ssgproject.content_profile_')[1];
+    }
+
+    rulesWithRemediations = (rules, system_id) => {
+        return window.insights.chrome.auth.getUser()
+        .then(() => {
+            return fetch('/api/remediations/v1/resolutions', {
+                method: 'POST',
+                headers: { 'Content-Type': 'application/json; chartset=utf-8' },
+                body: JSON.stringify({
+                    issues: rules.map(rule => `ssg:rhel7|` +
+                                      `${this.removeRefIdPrefix(rule.profiles[0].ref_id)}|` +
+                                      `${rule.ref_id}`)
+                })
+            }).then((response) => {
+                if (!response.ok) {
+                    // If remediations doesn't respond, inject no fix available
+                    return {};
+                }
+
+                return response.json();
+            }).then(response => Object.keys(response).filter(rule => response[rule]).map(
+                rule_ref_id => this.formatRule(this.findRule(rules, rule_ref_id), rule_ref_id.split('|')[1], system_id)
+            ));
+        });
+    }
+
     dataProvider = () => {
         const { allSystems, selectedRules } = this.props;
-        return allSystems.reduce((acc, { id, rule_objects_failed }) => ({
-            systems: [ ...acc.systems, id ],
-            issues: [
-                ...acc.issues,
-                ...selectedRules ?
-                    selectedRules.map(rule => this.formatRule(rule, id)) :
-                    rule_objects_failed.map(rule => this.formatRule(rule, id))
-            ]
-        }), { issues: [], systems: []});
-        /* eslint-enable camelcase */
+        const result = { systems: [], issues: []};
+        allSystems.forEach(async (system) => {
+            result.systems.push(system.id);
+            if (selectedRules) {
+                result.issues.push(selectedRules.map(rule => this.formatRule(rule, this.removeRefIdPrefix(rule.profile), system.id)));
+            } else {
+                result.issues.push(this.rulesWithRemediations(system.rule_objects_failed, system.id));
+            }
+        });
+
+        return Promise.all(result.issues).then(issues => {
+            result.issues = this.uniqIssuesBySystem(flatten(issues));
+            return result;
+        });
     }
+    /* eslint-enable camelcase */
 
     render() {
-        const { disableRemediations } = this.props;
+        const { allSystems, selectedRules } = this.props;
+
         return (
             <React.Fragment>
                 <RemediationButton
-                    isDisabled={ disableRemediations || this.dataProvider().issues.length === 0 }
+                    isDisabled={ (allSystems.length === 0 || allSystems[0].rule_objects_failed.length === 0) &&
```

Comment:
This condition looks suspicious - why checking the first system? Should not there be disjunction instead of conjunction?

Reply:
I basically copied over the ComplianceRemediationButton in `compliance-frontend` to here. In here, the button is used at a page where there's only one host